### PR TITLE
Panos commit

### PR DESCRIPTION
**Pull request by ivanbojer**
ISSUE TYPE

New Module Pull Request
COMPONENT NAME

panos_commit

ANSIBLE VERSION

ansible 2.1.0.0
  config file = 
  configured module search path = Default w/o overrides
SUMMARY

New panos module that allows user to commit current candidate configuration to the firewall

### CODE REVIEW

**Comment by ivanbojer** [comment]
ready_for_review

**Comment by gregdek** [comment]
Thanks @ivanbojer. To the current maintainers, @ivanbojer please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

File: network/panos/panos_commit.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+requirements:
+    - pan-python
+options:
+    ip_address:
+        description:
+            - IP address (or hostname) of PAN-OS device
+        required: true
+    password:
+        description:
+            - password for authentication
```
Comment (gundalow) [inline network/panos/panos_commit.py:40]: `descriptions` need to be sentences (e.g. capital letters & full stops)

File: network/panos/panos_commit.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        required: true
+    username:
+        description:
+            - username for authentication
```
Comment (gundalow) [inline network/panos/panos_commit.py:44]: Does the user need any specific permissions (if so detail), or is their only one level of user on PaloAltoNetworks equipment?

File: network/panos/panos_commit.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+        required: false
+        default: "admin"
+    interval:
+        description:
+            - interval for checking commit job
```
Comment (gundalow) [inline network/panos/panos_commit.py:49]: Please specify units
Interval (in seconds)..

File: network/panos/panos_commit.py [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+        required: false
+        default: 0.5
+    timeout:
+        description:
+            - timeout for commit job
```
Comment (gundalow) [inline network/panos/panos_commit.py:54]: Please specify units

File: network/panos/panos_commit.py [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+try:
+    import pan.xapi
+    HAS_LIB = True
+except ImportError:
+    HAS_LIB = False
+
+
+def main():
+    argument_spec = dict(
+        ip_address=dict(),
```
Comment (gundalow) [inline network/panos/panos_commit.py:92]: Please add `required=True,` where needed

File: network/panos/panos_commit.py [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+        timeout=dict(),
+        sync=dict(type='bool', default=True)
+    )
+    module = AnsibleModule(argument_spec=argument_spec, supports_check_mode=False)
+
+    if not HAS_LIB:
+        module.fail_json(msg='pan-python required for this module')
+
+    ip_address = module.params["ip_address"]
+    if not ip_address:
```
Comment (gundalow) [inline network/panos/panos_commit.py:105]: If you add the `required=True,` into the `argument_spec` section you can remove these checks

**Review by gundalow — CHANGES_REQUESTED** [review]
needs_revision

**Comment by gundalow** [comment]
Thanks for the module.

Not sure if you aware, we have a Network Working Group, check out https://github.com/ansible/community/blob/master/MEETINGS.md for details

1) What is needed to make this module idempotent (e.g. support check mode)
http://docs.ansible.com/ansible/dev_guide/developing_modules.html#check-mode

2) For us to accept new (or updated) network modules into 2.3 we require functional test to be added
https://github.com/ansible/ansible/blob/devel/test/integration/README.md#network-tests has more details.

If you need help with this feel free to message me directly `gundalow`  on FreeNode IRC. I'm in the UK, though IRC is always running, so feel free to leave me a message.

**Pull request closed without merging**